.generator/src/generator/templates/client.j2

```diff
@@ -115,35 +117,89 @@ func ParameterToString(obj interface{}, collectionFormat string) string {
 {# The method is used in Terraform client and needs to be public. -#}
 // CallAPI do the request.
 func (c *APIClient) CallAPI(request *http.Request) (*http.Response, error) {
-	if c.Cfg.Debug {
-		dump, err := httputil.DumpRequestOut(request, true)
-		if err != nil {
-			return nil, err
+	var rawBody []byte
+	if request.Body != nil && request.Body != http.NoBody {
+		rawBody, _ = io.ReadAll(request.Body)
+		request.Body.Close()
+	}
+
+	ctx, ccancel := context.WithTimeout(request.Context(), c.Cfg.RetryConfiguration.HTTPRetryTimeout)
+	defer ccancel()
+	maxRetries := c.Cfg.RetryConfiguration.MaxRetries
+	retryCount := 0
+	for {
+		if retryCount == maxRetries {
+			ccancel()
```

Comment:
Is that the right behavior? I would expect the same as the `!shouldRetry` below.

Reply:
If max retry is reached then next request is cancelled and we can exit without returning a response. Which is not a condition covered by shouldRetryRequest.

Reply:
AFAIU this will impact the next select, not the request itself? I think it would make sense to handle this in shouldRetryRequest.

Reply:
Yep I moved the logic to shouldRetryRequest